symfony/plugins/orangehrmTimePlugin/Report/AttendanceReport.php

```diff
@@ -0,0 +1,216 @@
+use OrangeHRM\Core\Traits\UserRoleManagerTrait;
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\Time\Dto\AttendanceReportSearchFilterParams;
+
+class AttendanceReport implements EndpointAwareReport
+{
+    use UserRoleManagerTrait;
+    use TextHelperTrait;
+
+    public const PARAMETER_EMPLOYEE_NAME = 'employeeName';
+    public const PARAMETER_TIME = 'time';
+
+    public const FILTER_EMP_NUMBER = 'empNumber';
+    public const FILTER_JOB_TITLE_ID = 'jobTitleId';
+    public const FILTER_SUBUNIT_ID = 'subunitId';
+    public const FILTER_EMPLOYMENT_STATUS_ID = 'employmentStatusId';
+    public const FILTER_PARAMETER_DATE_FROM = 'fromDate';
+    public const FILTER_PARAMETER_DATE_TO = 'toDate';
+
+
+    public const DEFAULT_COLUMN_SIZE = 150;
+
+    /**
+     * @inheritDoc
+     */
+    public function prepareFilterParams(EndpointProxy $endpoint): FilterParams
+    {
+        $filterParams = new AttendanceReportSearchFilterParams();
+
+        $endpoint->setSortingAndPaginationParams($filterParams);
+
+        $empNumber = $endpoint->getRequestParams()->getIntOrNull(
+            RequestParams::PARAM_TYPE_QUERY,
+            self::FILTER_EMP_NUMBER
+        );
+
+        if (!is_null($empNumber)) {
+            $filterParams->setEmployeeNumbers([$empNumber]);
+        } else {
+            $accessibleEmpNumbers = $this->getUserRoleManager()->getAccessibleEntityIds(Employee::class);
+            $filterParams->setEmployeeNumbers($accessibleEmpNumbers);
+        }
+
+        $filterParams->setJobTitleId(
+            $endpoint->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_JOB_TITLE_ID
+            )
+        );
+        $filterParams->setSubunitId(
+            $endpoint->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_SUBUNIT_ID
+            )
+        );
+        $filterParams->setEmploymentStatusId(
+            $endpoint->getRequestParams()->getIntOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_EMPLOYMENT_STATUS_ID
+            )
+        );
+        $filterParams->setFromDate(
+            $endpoint->getRequestParams()->getDateTimeOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_PARAMETER_DATE_FROM
+            )
+        );
+        $filterParams->setToDate(
+            $endpoint->getRequestParams()->getDateTimeOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_PARAMETER_DATE_TO
+            )
+        );
+
+        return $filterParams;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRule(EndpointProxy $endpoint): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $endpoint->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    CommonParams::PARAMETER_EMP_NUMBER,
```

Comment:
in this class you have `self::FILTER_EMP_NUMBER`

Reply:
done